### PR TITLE
PoC: Remove global version variable; separate out version helpers

### PR DESCRIPTION
**Pull request by erlend-aasland**
Low-hanging fruit 🍏

### CODE REVIEW

**Comment by erlend-aasland** [comment]
Added possible enhancement in 420b8e582716dc8bc9b6d7f6960b3ab6cb009364: move stateless version helpers to a `version.py` file under a _libclinic_ package.

File: Tools/clinic/clinic.py [hunk 1]
```diff
@@ -50,6 +50,11 @@
     overload,
 )
 
+
+# Local includes.
+import libclinic
```
Comment (AlexWaygood) [inline Tools/clinic/clinic.py:55]: did you forget to `git add` this file

Comment (erlend-aasland) [reply Tools/clinic/clinic.py:55]: Probably!

Comment (erlend-aasland) [reply Tools/clinic/clinic.py:55]: git added

File: Tools/clinic/clinic.py [hunk 2]
```diff
@@ -5242,11 +5204,13 @@ def reset(self) -> None:
...
-            fail("Insufficient Clinic version!\n"
-                 f"  Version: {version}\n"
-                 f"  Required: {required}")
+        try:
+            if libclinic.version_comparator(self.VERSION, required) < 0:
+                fail("Insufficient Clinic version!\n"
+                     f"  Version: {self.VERSION}\n"
+                     f"  Required: {required}")
+        except libclinic.ParseVersionError as err:
+            fail(str(err))
```
Comment (AlexWaygood) [inline Tools/clinic/clinic.py:5213]: It feels like, both before and after this PR, we'd have quite a lot of code to support this feature. The feature doesn't seem to be documented at all. It was added in https://github.com/python/cpython/commit/ebdcb50b8a0d37af4acd7d2387eae8ff2b5f0b9b, ten years ago. It has one test (which we added thsi year, as part of an effort to increase overall test coverage for `clinic.py`.

Is the feature used? Has it ever been used? What are the situations in which we might use it?

Maybe we could just... remove it 😈

Comment (erlend-aasland) [reply Tools/clinic/clinic.py:5213]: Yeah, I was thinking the same. It has never been used. It _could_ be useful if we chose to expose Argument Clinic as an official tool, but we could easily add it back if we chose to go down that path.

Comment (AlexWaygood) [reply Tools/clinic/clinic.py:5213]: > It _could_ be useful if we chose to expose Argument Clinic as an official tool, but we could easily add it back if we chose to go down that path.

Yeah, and that would be a really significant undertaking that I don't even want to _think_ about until we've got the code to a cleaner, more presentable state :) I feel like we shouldn't be making decisions now on the basis of "Well, we might change the whole purpose of this codebase at a later date". So I'd support just ripping the feature out :)

Comment (erlend-aasland) [reply Tools/clinic/clinic.py:5213]: Let's do it!

Comment (erlend-aasland) [reply Tools/clinic/clinic.py:5213]: See https://github.com/python/cpython/issues/113336

**Pull request closed without merging**